### PR TITLE
Project Attachments: Destroy

### PR DESCRIPTION
**Pull request by ChrisHuynh333**
## What does this PR do and why?
This PR adds file deletion to project attachments.

## Screenshots or screen recordings
[Screencast from 2024-09-17 02:21:17 PM.webm](https://github.com/user-attachments/assets/40f5a99e-ac00-423a-942c-cc6561d4c48c)

## How to set up and validate locally
1. Navigate to a project `Files` page.
2. Upload files if needed (including paired end files)
3. Delete the files using the `Delete` link within each row (paired end files should both delete with a single action)
4. Ensure the table updates and no longer includes the deleted file(s)

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/views/projects/attachments/_delete_attachment_modal.html.erb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    <div
+      class="mb-4 font-normal text-slate-500 dark:text-slate-400 overflow-x-visible">
+      <p class="mb-4">
+        <%= t(".description") %>
+      </p>
+      <%= form_for(:deletion, url: namespace_project_attachment_path(id: @attachment.id), method: :delete) do |form| %>
+        <%= form.submit t(".submit_button"),
+                    class: "
+                      button
```
Comment (joshsadam) [inline app/views/projects/attachments/_delete_attachment_modal.html.erb:15]: **nitpick:** Don;t use those pre-defined classes, I am trying to get rid of them

**Review by joshsadam — CHANGES_REQUESTED** [review]
Just the one comment on not using the `button` class, the rest works as expected and looks good.  Nice work.

File: app/views/projects/attachments/_delete_attachment_modal.html.erb (hunk 1 above)
Comment (ChrisHuynh333) [reply app/views/projects/attachments/_delete_attachment_modal.html.erb]: Replaced in 604de82cfeb991875d5ebf9263de9739a7610a6d

File: app/controllers/projects/attachments_controller.rb [hunk 2]
```diff
@@ -44,6 +46,36 @@ def create
...
+                                               }), status: :ok
+    end
+
+    def destroy # rubocop:disable Metrics/MethodLength
+      @destroyed_attachments = ::Attachments::DestroyService.new(@project.namespace, @attachment, current_user).execute
+
+      respond_to do |format|
+        if @destroyed_attachments.count.positive?
+          status = destroy_status(@attachment, @destroyed_attachments.length)
+          format.turbo_stream do
```
Comment (deepsidhu85) [inline app/controllers/projects/attachments_controller.rb:64]: So overall this PR is great and working as expected. The message needs to be updated. If you upload a pair of files and then delete them the text in the modal reads:

image

The message makes sense when you are deleting a single file but here 2 files will be removed

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This looks great just have one comment below to update the text in the delete modal

File: app/controllers/projects/attachments_controller.rb (hunk 2 above)
Comment (ChrisHuynh333) [reply app/controllers/projects/attachments_controller.rb:64]: Fixed in 51bd2bb70e420b759680a25480266a9f73522d89 / 37e394d5d3e09568617c102666274dd1c9c089c7

Added filename(s) to the modal as well
image

Comment (deepsidhu85) [reply app/controllers/projects/attachments_controller.rb:64]: Perfect thanks!

**Review by deepsidhu85 — APPROVED** [review]
Looks good to me!

**Review by joshsadam — APPROVED** [review]

**Pull request merged**